Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $AppServiceName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $appService = Get-AzWebApp -ResourceGroupName $ResourceGroupName -Name $AppServiceName
+
+  $ftpConfigs = [ordered]@{ "Production" = $appService.SiteConfig.FtpsState }
+
+  # List of slots in Get-AzWebAppSlot has full slot name. SiteConfig is blank on those.
+  # So we have to get just the bare slot name, then call Get-AzWebAppSlot for it to get the SiteConfig and FtpsState.
+  ForEach ($slotNameRaw in (Get-AzWebAppSlot -ResourceGroupName $ResourceGroupName -Name $AppServiceName).Name)
+  {
+    $slotName = $slotNameRaw.Replace($AppServiceName + "/", "")
+    $slot = Get-AzWebAppSlot -ResourceGroupName $ResourceGroupName -Name $AppServiceName -Slot $slotName
+    $ftpConfigs = $ftpConfigs + @{ $slotName = $slot.SiteConfig.FtpsState }
+  }
+
+  return $ftpConfigs
+}
+
+function Set-AppServiceFtpState()
+{
+  <#
+    .SYNOPSIS
+    This command sets the FTP state for the App Service.
+    .DESCRIPTION
+    This command sets the FTP state for the App Service.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the App Service.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the App Service.
+    .PARAMETER AppServiceName
+    The App Service name.
+    .PARAMETER SlotName
+    If a non-production slot's FTP state is being set, provide the slot name. To set the default production slot FTP state, do not specify this parameter.
+    .PARAMETER FtpState
+    The FTP State to set. Either "Disabled" or "FtpsOnly". Defaults to "Disabled".
+    .INPUTS
+    None
+    .OUTPUTS
+    None
+    .EXAMPLE
+    PS> Set-AppServiceFtpState -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -AppServiceName "MyAppServiceName" -SlotName "MyNonProductionSlotName" -FtpState "Disabled"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $AppServiceName,
+    [Parameter(Mandatory = $false)]
+    [string]
+    $SlotName,
+    [Parameter(Mandatory = $false)]
+    [string]
+    $FtpState = "Disabled"
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  if ([string]::IsNullOrWhiteSpace($SlotName))
+  {
+    # No slot name so we're setting the app service itself
+    Set-AzWebApp -ResourceGroupName $ResourceGroupName -Name $AppServiceName -FtpsState $FtpState
+  }
+  else
+  {
+    if ($SlotName.StartsWith($AppServiceName))
```

Comment:
Corner case, but if some user select slot name which in actual starts with AppService name, then this logic might misbehave.

Reply:
Updating logic to check for app service name plus forward slash, as that is how Get-AzWebAppSlot returns slot names, and that should mitigate possible corner case where slot name starts with app svc name.